doc/README.md

```diff
@@ -0,0 +1,29 @@
+# Documentation Index
+
+The following is a list of documentation files and a brief description to aid
+in locating documentation (sorted alphabetically).
+
+## Lua
+
+- [Builtin Entities](builtin_entities.md): Documentation of core engine entities.
```

Comment:
These "descriptions" are all very redundant by being of the form "X: Documentation of X"

IMO please add a proper description, or omit the description entirely; but in the latter case, I would be inclined to agree with SmallJoker that this doesn't add that much over the file names?